gpiozero/output_devices.py

```diff
@@ -300,7 +299,10 @@ def __init__(self, pin=None, active_high=True, initial_value=0, frequency=100):
         self._controller = None
         if not 0 <= initial_value <= 1:
             raise OutputDeviceBadValue("initial_value must be between 0 and 1")
-        super(PWMOutputDevice, self).__init__(pin, active_high)
+        if initial_value in (0, 1):
+            super(PWMOutputDevice, self).__init__(pin, active_high, initial_value=initial_value)
```

Comment:
@bennuttall At the moment the pin classes have some "utility" methods like `output_with_state` to facilitate configuring pins in a given state as fast as possible (only possible with some underlying implementations but helps prevent "power blips" on component construction - in some use-cases this is important). However, there currently isn't an equivalent facility for the PWM side of things - so setting pin frequency always results in the pin switching off (however briefly) at the moment. I need to add another utility method for this.